src/compiler/checker.ts

```diff
@@ -5823,6 +5823,20 @@ namespace ts {
                 return !(getObjectFlags(type) & ObjectFlags.Reference) || !isTypeReferenceNode(existing) || length(existing.typeArguments) >= getMinTypeArgumentCount((type as TypeReference).target.typeParameters);
             }
 
+            function serializeParameterTypeForDeclaration(context: NodeBuilderContext, type: Type, symbol: Symbol, enclosingDeclaration: Node | undefined, includePrivateSymbol?: (s: Symbol) => void, bundled?: boolean) {
+                if (type !== errorType && symbol.valueDeclaration?.kind === SyntaxKind.Parameter && (symbol.valueDeclaration as ParameterDeclaration).questionToken) {

```

Comment:
Do we have to look at the value declaration here? Or is there a `CheckFlag` on the symbol which indicates optionality? Moreover, do we already have an `isOptionalParameterSymbol` helper? I can't remember offhand.